go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,299 @@
+	keyspace string
+
+	// Host IP addresses of the cluster
+	hosts []string
+
+	config *registry.RepoConfig
+}
+
+func NewCassandraOnlineStore(project string, config *registry.RepoConfig, onlineStoreConfig map[string]interface{}) (*CassandraOnlineStore, error) {
+	store := CassandraOnlineStore{
+		project: project,
+		config:  config,
+	}
+
+	// Parse host_name and Ips
+	cassandraHosts, ok := onlineStoreConfig["hosts"]
+	if !ok {
+		cassandraHosts = []interface{}{"127.0.0.1"}
+		log.Warn().Msg("host not provided: Using 127.0.0.1 instead")
+	}
+
+	var rawCassandraHosts []interface{}
+	if rawCassandraHosts, ok = cassandraHosts.([]interface{}); !ok {
+		return nil, fmt.Errorf("didn't pass a list of hosts in the 'hosts' field")
+	}
+
+	var cassandraHostsStr = make([]string, len(rawCassandraHosts))
+	for i, rawHost := range rawCassandraHosts {
+		hostStr, ok := rawHost.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert a host to a string: %+v", rawHost)
+		}
+		cassandraHostsStr[i] = hostStr
+	}
+
+	username, ok := onlineStoreConfig["username"]
+	if !ok {
+		username = "cassandra"
+		log.Warn().Msg("username not defined: Using default username instead")
+	}
+
+	password, ok := onlineStoreConfig["password"]
+	if !ok {
+		password = "cassandra"
+		log.Warn().Msg("password not defined: Using default password instead")
+	}
+
+	var usernameStr string
+	if usernameStr, ok = username.(string); !ok {
+		return nil, fmt.Errorf("failed to convert username to string: %+v", usernameStr)
+	}
+
+	var passwordStr string
+	if passwordStr, ok = password.(string); !ok {
+		return nil, fmt.Errorf("failed to convert password to string: %+v", passwordStr)
+	}
+
+	keyspace, ok := onlineStoreConfig["keyspace"]
+	if !ok {
+		keyspace = "scylladb"
+		log.Warn().Msg("Keyspace not defined: Using 'scylladb' as keyspace instead")
+	}
+	store.keyspace = keyspace.(string)
+
+	var keyspaceStr string
+	if keyspaceStr, ok = keyspace.(string); !ok {
+		return nil, fmt.Errorf("failed to convert keyspace to string: %+v", keyspaceStr)
+	}
+
+	protocolVersion, ok := onlineStoreConfig["protocol_version"]
+	if !ok {
+		protocolVersion = 4.0
+		log.Warn().Msg("protocol_version not specified: Using 4 instead")
+	}
+	protocolVersionInt := int(protocolVersion.(float64))
+
+	redisTraceServiceName := os.Getenv("DD_SERVICE") + "-cassandra"
+	if redisTraceServiceName == "" {
+		redisTraceServiceName = "cassandra.client" // default service name if DD_SERVICE is not set
+	}
+	store.clusterConfigs = gocqltrace.NewCluster(cassandraHostsStr, gocqltrace.WithServiceName(redisTraceServiceName))
+	// TODO: Figure out if we need to offer users the ability to tune the timeouts
+	//store.clusterConfigs.ConnectTimeout = 1
+	//store.clusterConfigs.Timeout = 1
+	store.clusterConfigs.ProtoVersion = protocolVersionInt
+	//store.clusterConfigs.Consistency = gocql.Quorum
+	store.clusterConfigs.Keyspace = keyspaceStr
+	loadBalancingPolicy, ok := onlineStoreConfig["load_balancing"]
```

Comment:
load_balancing object has load_balancing_policy and local_dc fields. Seems they are not parsed correctly.

Reply:
Yeah, that's right, addressed this.